src/Faction/Faction.ts

```diff
@@ -75,6 +79,30 @@ export class Faction {
     return newFavor - this.favor;
   }
 
+  checkForInvite(p: PlayerObject): boolean {
+    if (this.isBanned) return false;
+    if (this.isMember) return false;
+    if (this.alreadyInvited) return false;
+    const conditions = this.getInfo().inviteReqs;
+    if (conditions.length == 0) return false;
+    for (const condition of conditions) {
+      if (!condition.isSatisfied(p)) return false;
+    }
+    return true;
+  }
+
+  checkForRumor(p: PlayerObject): boolean {
+    if (this.isBanned) return false;
+    if (this.isMember) return false;
+    if (this.alreadyInvited) return false;
+    const conditions = this.getInfo().rumorReqs;
+    if (conditions.length == 0) return false;
+    for (const condition of conditions) {
+      if (!condition.isSatisfied(p)) return false;
+    }
+    return true;
+  }
+
```

Comment:
Neither of these functions appear to be used anywhere, is there any issue with getting rid of these?

Reply:
These are used in PlayerObjectGeneralMethods `checkForFactionInvitations()` and replace all of the logic that was in that function. They could maybe be inlined there to avoid the redundant checks for bans, but I like how this implementation separates the side-effect-free code (inviites are given one at a time) from the side-effect code (rumors are given in bulk and never cause popups).

Reply:
Oh weird, I don't know why vscode didn't see that when I checked references the first time, I see it now though.

I'll get this fully reviewed tonight/early morning (east coast US) and hopefully merged. For minor gripes I'll probably just make some small changes before merging and I'll just let you know what was changed and why.